### PR TITLE
feat: 11830 performance improvements log fwk

### LINKED ISSUE #11830: Performance Improvements on Swirlds-Logging
We identified some performance improvements to be introduced in the logging framework.

- OutputStream instead of BufferedWriter.
- ByteBuffer instead of StringBuffer.
- StringBuilder instead of StringBuffer in singled-threaded code
- synchronized blocks instead of locks.

This task is also introducing the swirls-logging-benchmark module to be able to run benchmarks against future improvements.

Pending, find the configuration for bufferSize that maximizes throughput.
 
Related:#11740

### PR DESCRIPTION
**Pull request by mxtartaglia-sl**
Fixes #11830

Result from lastest jmh performance testing 

```
Benchmark                                    (loggingType)   Mode  Cnt      Score       Error  Units
Log4JBenchmark.log4J                               CONSOLE  thrpt    4    417.589 ±   143.913  ops/s
Log4JBenchmark.log4J                                  FILE  thrpt    4  19635.480 ±  1211.899  ops/s
Log4JBenchmark.log4J                      CONSOLE_AND_FILE  thrpt    4    268.843 ±   253.720  ops/s
SwirldsLogBenchmark.swirldsLogging                 CONSOLE  thrpt    4    299.232 ±   660.481  ops/s
SwirldsLogBenchmark.swirldsLogging                    FILE  thrpt    4  35849.778 ± 10177.849  ops/s
SwirldsLogBenchmark.swirldsLogging        CONSOLE_AND_FILE  thrpt    4    275.705 ±   141.983  ops/s
```

### CODE REVIEW

File: platform-sdk/swirlds-logging/build.gradle.kts [hunk 1]
```diff
@@ -38,3 +33,22 @@ testModuleInfo {
...
+    requires("com.swirlds.logging")
+    requires("org.apache.logging.log4j")
+    requires("com.swirlds.config.api")
+    runtimeOnly("com.swirlds.config.impl")
+    requires("org.apache.logging.log4j.core")
+    requires("com.github.spotbugs.annotations")
+    requires("jmh.core")
+}
+
+// IntelliJ uses adhoc-created JavaExec tasks when running a 'main()' method.
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/build.gradle.kts:47]: @jjohannes @nathanklick looks like we need that block whenever we want to exec a main. Can we add that to a plugin?

File: platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureLog4J.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+public class ConfigureLog4J {
+
+    private static final String PATTERN = "%d{UNIX_MILLIS} %-5level [%t] %c - %msg - [%marker] %X %n%throwable";
+    public static final String CONSOLE_APPENDER_NAME = "console";
+    public static final String FILE_APPENDER_NAME = "file";
+
+    public static @NonNull LoggerContext configureConsoleLogging() {
+        System.clearProperty("log4j2.contextSelector");
+        ConfigurationBuilder<BuiltConfiguration> builder = ConfigurationBuilderFactory.newConfigurationBuilder();
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureLog4J.java:43]:
```suggestion
        final ConfigurationBuilder<BuiltConfiguration> builder = ConfigurationBuilderFactory.newConfigurationBuilder();
```

File: platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureLog4J.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        builder.setConfigurationName("consoleLoggingConfig");
+        builder.add(createConsoleAppender(builder));
+        builder.add(builder.newRootLogger(Level.DEBUG).add(builder.newAppenderRef(CONSOLE_APPENDER_NAME)));
+        return create(builder);
+    }
+
+    public static @NonNull LoggerContext configureFileLogging() {
+        final String logFile = LogFiles.provideLogFilePath(LoggingImplementation.LOG4J2, LoggingHandlingType.FILE);
+        System.clearProperty("log4j2.contextSelector");
+        ConfigurationBuilder<BuiltConfiguration> builder = ConfigurationBuilderFactory.newConfigurationBuilder();
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureLog4J.java:54]:
```suggestion
        final ConfigurationBuilder<BuiltConfiguration> builder = ConfigurationBuilderFactory.newConfigurationBuilder();
```

File: platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureLog4J.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        builder.add(createFileAppender(builder, logFile));
+        builder.add(builder.newRootLogger(Level.DEBUG).add(builder.newAppenderRef(FILE_APPENDER_NAME)));
+        return create(builder);
+    }
+
+    public static @NonNull LoggerContext configureFileAndConsoleLogging() {
+        final String logFile =
+                LogFiles.provideLogFilePath(LoggingImplementation.LOG4J2, LoggingHandlingType.CONSOLE_AND_FILE);
+        System.clearProperty("log4j2.contextSelector");
+        ConfigurationBuilder<BuiltConfiguration> builder = ConfigurationBuilderFactory.newConfigurationBuilder();
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureLog4J.java:66]:
```suggestion
        final ConfigurationBuilder<BuiltConfiguration> builder = ConfigurationBuilderFactory.newConfigurationBuilder();
```

File: platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureLog4J.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        builder.setConfigurationName("fileAndConsoleLoggingConfig");
+        builder.add(createFileAppender(builder, logFile));
+        builder.add(createConsoleAppender(builder));
+        builder.add(builder.newRootLogger(Level.DEBUG)
+                .add(builder.newAppenderRef(FILE_APPENDER_NAME))
+                .add(builder.newAppenderRef(CONSOLE_APPENDER_NAME)));
+        return create(builder);
+    }
+
+    private static @NonNull LoggerContext create(ConfigurationBuilder<BuiltConfiguration> builder) {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureLog4J.java:77]:
```suggestion
    private static @NonNull LoggerContext create(@NonNull final ConfigurationBuilder<BuiltConfiguration> builder) {
```

File: platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureLog4J.java [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        Configuration configuration = builder.build();
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureLog4J.java:78]: missing null check + final

File: platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureLog4J.java [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+import org.apache.logging.log4j.core.config.Configuration;
+import org.apache.logging.log4j.core.config.Configurator;
+import org.apache.logging.log4j.core.config.builder.api.AppenderComponentBuilder;
+import org.apache.logging.log4j.core.config.builder.api.ConfigurationBuilder;
+import org.apache.logging.log4j.core.config.builder.api.ConfigurationBuilderFactory;
+import org.apache.logging.log4j.core.config.builder.api.LayoutComponentBuilder;
+import org.apache.logging.log4j.core.config.builder.impl.BuiltConfiguration;
+import org.apache.logging.log4j.spi.LoggerContext;
+
+public class ConfigureLog4J {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureLog4J.java:35]: Let's add some Java doc to the class

File: platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureLog4J.java [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+    private static final String PATTERN = "%d{UNIX_MILLIS} %-5level [%t] %c - %msg - [%marker] %X %n%throwable";
+    public static final String CONSOLE_APPENDER_NAME = "console";
+    public static final String FILE_APPENDER_NAME = "file";
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureLog4J.java:39]: why public?

File: platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureLog4J.java [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+
+    private static @NonNull LoggerContext create(ConfigurationBuilder<BuiltConfiguration> builder) {
+        Configuration configuration = builder.build();
+        org.apache.logging.log4j.core.LoggerContext context = Configurator.initialize(configuration);
+        LogManager.getFactory().removeContext(context);
+        return Configurator.initialize(configuration);
+    }
+
+    private static AppenderComponentBuilder createConsoleAppender(
+            final ConfigurationBuilder<BuiltConfiguration> builder) {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureLog4J.java:85]: nonnull + null check

File: platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureLog4J.java [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+        final LayoutComponentBuilder layoutComponentBuilder =
+                builder.newLayout("PatternLayout").addAttribute("pattern", PATTERN);
+        return builder.newAppender(ConfigureLog4J.CONSOLE_APPENDER_NAME, "CONSOLE")
+                .addAttribute("target", ConsoleAppender.Target.SYSTEM_OUT)
+                .add(layoutComponentBuilder);
+    }
+
+    private static AppenderComponentBuilder createFileAppender(
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureLog4J.java:93]: nonnull + null check

File: platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureSwirldsLog.java [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+import com.swirlds.logging.api.internal.LoggingSystem;
+import com.swirlds.logging.api.internal.configuration.ConfigLevelConverter;
+import com.swirlds.logging.api.internal.configuration.MarkerStateConverter;
+import com.swirlds.logging.benchmark.config.LoggingHandlingType;
+import com.swirlds.logging.benchmark.config.LoggingImplementation;
+import com.swirlds.logging.benchmark.util.LogFiles;
+import com.swirlds.logging.external.benchmark.BenchmarkFactory;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+public class ConfigureSwirldsLog {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureSwirldsLog.java:31]: Add javadoc to class

File: platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureSwirldsLog.java [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+                .withConverter(new MarkerStateConverter())
+                .withValue("logging.level", "trace")
+                .withValue("logging.handler.file.type", "file")
+                .withValue("logging.handler.file.active", "true")
+                .withValue("logging.handler.file.formatTimestamp", "false")
+                .withValue("logging.handler.file.level", "trace")
+                .withValue("logging.handler.file.file", logFile)
+                .build();
+        final LogHandler fileHandler = BenchmarkFactory.getFileHandlerFactory().create("file", configuration);
+        LoggingSystem loggingSystem = new LoggingSystem(configuration);
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureSwirldsLog.java:46]:
```suggestion
        final LoggingSystem loggingSystem = new LoggingSystem(configuration);
```

File: platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureSwirldsLog.java [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+                .withConverter(new MarkerStateConverter())
+                .withValue("logging.level", "trace")
+                .withValue("logging.handler.console.type", "console")
+                .withValue("logging.handler.console.active", "true")
+                .withValue("logging.handler.console.formatTimestamp", "false")
+                .withValue("logging.handler.console.level", "trace")
+                .build();
+        final LogHandler consoleHandler =
+                BenchmarkFactory.getConsoleHandlerFactory().create("console", configuration);
+        LoggingSystem loggingSystem = new LoggingSystem(configuration);
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureSwirldsLog.java:63]:
```suggestion
        final LoggingSystem loggingSystem = new LoggingSystem(configuration);
```

File: platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureSwirldsLog.java [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+                .withValue("logging.handler.file.file", logFile)
+                .withValue("logging.handler.console.type", "console")
+                .withValue("logging.handler.console.active", "true")
+                .withValue("logging.handler.console.formatTimestamp", "false")
+                .withValue("logging.handler.console.level", "trace")
+                .build();
+        final LogHandler fileHandler = BenchmarkFactory.getFileHandlerFactory().create("file", configuration);
+        final LogHandler consoleHandler =
+                BenchmarkFactory.getConsoleHandlerFactory().create("console", configuration);
+        LoggingSystem loggingSystem = new LoggingSystem(configuration);
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/ConfigureSwirldsLog.java:88]:
```suggestion
        final LoggingSystem loggingSystem = new LoggingSystem(configuration);
```

File: platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/LogWithLog4J.java [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+package com.swirlds.logging.benchmark;
+
+import com.swirlds.logging.benchmark.util.BenchmarkUtils;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.Marker;
+import org.apache.logging.log4j.MarkerManager;
+import org.apache.logging.log4j.ThreadContext;
+
+public class LogWithLog4J implements Runnable {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/LogWithLog4J.java:26]: java doc to class

File: platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/LogWithSwirlds.java [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+ * limitations under the License.
+ */
+
+package com.swirlds.logging.benchmark;
+
+import com.swirlds.logging.api.Level;
+import com.swirlds.logging.api.Logger;
+import com.swirlds.logging.benchmark.util.BenchmarkUtils;
+
+public class LogWithSwirlds implements Runnable {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/LogWithSwirlds.java:23]: java doc to class

File: platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/util/BenchmarkUtils.java [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.logging.benchmark.util;
+
+import java.util.UUID;
+
+public class BenchmarkUtils {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/util/BenchmarkUtils.java:21]: javadoc

File: platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/util/LogFiles.java [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+package com.swirlds.logging.benchmark.util;
+
+import com.swirlds.logging.benchmark.config.LoggingHandlingType;
+import com.swirlds.logging.benchmark.config.LoggingImplementation;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public class LogFiles {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/util/LogFiles.java:26]: javadoc

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/Level.java [hunk 19]
```diff
@@ -84,4 +84,22 @@ public static Level valueOfOrElse(@Nullable final String value, @NonNull final L
             return defaultLevel;
         }
     }
+
+    public String nameWithFixedSize() {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/Level.java:88]: nonnull + javadoc

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/extensions/handler/AbstractAsyncHandler.java [hunk 20]
```diff
@@ -0,0 +1,81 @@
...
+package com.swirlds.logging.api.extensions.handler;
+
+import com.swirlds.config.api.Configuration;
+import com.swirlds.logging.api.extensions.event.LogEvent;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public abstract class AbstractAsyncHandler extends AbstractLogHandler {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/extensions/handler/AbstractAsyncHandler.java:26]: I ask myself if we should remove that class for now

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/format/DateFormatUtils.java [hunk 21]
```diff
@@ -0,0 +1,113 @@
...
+
+import com.swirlds.logging.api.Level;
+import com.swirlds.logging.api.extensions.emergency.EmergencyLogger;
+import com.swirlds.logging.api.extensions.emergency.EmergencyLoggerProvider;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+
+public class DateFormatUtils {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/format/DateFormatUtils.java:27]: javadoc

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/format/LineBasedFormat.java [hunk 22]
```diff
@@ -24,33 +27,33 @@
...
     private static final EmergencyLogger EMERGENCY_LOGGER = EmergencyLoggerProvider.getEmergencyLogger();
 
-    /**
-     * The formatter for the timestamp.
-     */
-    private static final DateTimeFormatter FORMATTER =
-            DateTimeFormatter.ISO_LOCAL_DATE_TIME.withZone(ZoneId.systemDefault());
+    private final boolean formatTimestamp;
+
+    public LineBasedFormat(boolean formatTimestamp) {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/format/LineBasedFormat.java:49]: javadoc

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/external/benchmark/BenchmarkFactory.java [hunk 23]
```diff
@@ -0,0 +1,38 @@
...
+
+import com.swirlds.logging.api.extensions.handler.LogHandlerFactory;
+import com.swirlds.logging.console.ConsoleHandlerFactory;
+import com.swirlds.logging.file.FileHandlerFactory;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Class that exposes the factories, so they can be used in the benchmarking module
+ */
+public class BenchmarkFactory {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/external/benchmark/BenchmarkFactory.java:27]: Can this class be moved to jmh src set?

Comment (mxtartaglia-sl) [reply platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/external/benchmark/BenchmarkFactory.java:27]: Oh, I remember; this class existed only for not having to individually export each package containing a factory to the benchmark when it was its module. Now, given that jmh source can access the files directly, I can do that or instantiate the factories without the need for this class.

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/format/LineBasedFormat.java (hunk 22 above)
Comment (mxtartaglia-sl) [reply platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/format/LineBasedFormat.java]: added + renamed the class for better stating its purpose

File: platform-sdk/swirlds-logging/src/test/java/com/swirlds/logging/buffer/BufferedOutputStreamTest.java [hunk 24]
```diff
@@ -0,0 +1,149 @@
...
+    }
+
+    @Test
+    void testWriteNoFlushArray() throws IOException {
+        // Given
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        try (BufferedOutputStream bufferedOutputStream = new BufferedOutputStream(outputStream, 7); ) {
+
+            // When / Then
+            final byte[] a = "Hello ".getBytes();
```
[the remark this answers is not in the document]
Comment (mxtartaglia-sl) [reply platform-sdk/swirlds-logging/src/test/java/com/swirlds/logging/buffer/BufferedOutputStreamTest.java]: Done, thanks for spotting for the miss

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/format/DateFormatUtils.java (hunk 21 above)
Comment (mxtartaglia-sl) [reply platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/format/DateFormatUtils.java]: added plus renamed for better state its logic

**Review by jeromy-cannon — APPROVED** [review]

**Review by nathanklick — APPROVED** [review]

File: platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/util/Throwables.java [hunk 25]
```diff
@@ -0,0 +1,67 @@
...
+
+package com.swirlds.logging.benchmark.util;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.UUID;
+
+/**
+ * Convenience methods for creating exceptions with stacktrace as big as requested
+ */
+public class Throwables {
```
[the remark this answers is not in the document]
Comment (mxtartaglia-sl) [reply platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/util/Throwables.java:25]: As we discussed, we are addressing this in a different PR and marking this as resolved to merge this one.

File: platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/util/ConfigManagement.java [hunk 26]
```diff
@@ -0,0 +1,61 @@
...
+package com.swirlds.logging.benchmark.util;
+
+import com.swirlds.logging.benchmark.config.Constants;
+
+/**
+ * Utility class for configuring benchmark handling of
+ * <li>timestamp formatting
+ * <li>outputFile deletion
+ */
+public class ConfigManagement {
```
[the remark this answers is not in the document]
Comment (mxtartaglia-sl) [reply platform-sdk/swirlds-logging/src/jmh/java/com/swirlds/logging/benchmark/util/ConfigManagement.java:26]: As we discussed, we are addressing this in a different PR and marking this as resolved to merge this one.

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/Level.java (hunk 19 above)
Comment (mxtartaglia-sl) [reply platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/Level.java:88]: As we discussed, we are addressing this in a different PR and marking this as resolved to merge this one.

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/extensions/handler/AbstractAsyncHandler.java (hunk 20 above)
Comment (mxtartaglia-sl) [reply platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/extensions/handler/AbstractAsyncHandler.java:26]: We definitely should. 
Addressing this in a different PR and marking this as resolved to merge this one.

**Pull request merged**